tests/tests/test_state_scripts.py

```diff
@@ -632,6 +764,139 @@ class TestStateScripts(MenderTesting):
         "ArtifactFailure_Error_55", # Error for this state doesn't exist, should never run.
     ]
 
+    @pytest.mark.usefixtures("standard_setup_one_client_bootstrapped")
+    @pytest.mark.parametrize("test_set", REBOOT_TEST_SET)
+    def test_reboot_recovery(self, test_set):
+        if not env.host_string:
+            execute(
+                self.test_reboot_recovery,
+                test_set,
+                hosts=get_mender_clients())
+            return
+
+        client = env.host_string
+        work_dir = "test_state_scripts.%s" % client
+
+        script_content = '#!/bin/sh\n\necho "$(basename $0)" >> /data/test_state_scripts.log\n'
+        # Even though this script is actually run twice, only show it once in the logs
+        script_stop_mender_on_entry = '#!/bin/sh\ntest "$(grep -c "$(basename $0)" /data/test_state_scripts.log)" -eq 0 && echo "$(basename $0)" >> /data/test_state_scripts.log && systemctl stop mender\nexit 0\n'
```

Comment:
I didn't get this. Why does it run twice?

Reply:
This is for the case of a powerloss in commit_leave, in which case the script is rerun after it has booted.

Reply:
Because mender does not stop and start right at the execution point where it dies, it will in this case start over again from reboot_leave, as it is still on the same  partition as when it died (no reboot has happened).

Reply:
Oh, so you're fooling it into doing the same thing twice.